src/data.js

```diff
@@ -20,6 +20,27 @@ define( [
 var rbrace = /^(?:\{[\w\W]*\}|\[[\w\W]*\])$/,
 	rmultiDash = /[A-Z]/g;
 
+function getData( data ) {
+	switch ( data ) {
+		case "true":
+			return true;
+		case "false":
+			return false;
+		case "null":
+			return null;
+
+		// Only convert to a number if it doesn't change the string
+		case ( +data + "" ):
+			return +data;
+		default:
+			if ( rbrace.test( data ) ) {
```

Comment:
I also wonder if changing that to `if-else if-else if` etc. wouldn't compress better.